exec/java-exec/src/main/java/org/apache/drill/exec/store/AbstractSchema.java

```diff
@@ -194,4 +195,26 @@ public void dropTable(String tableName) {
         .message("Dropping tables is not supported in schema [%s]", getSchemaPath())
         .build(logger);
   }
-}
+
+  /**
+   * Visit the tables in this schema and write to recordGenerator
+   * @param recordGenerator recordGenerator for the output
+   * @param schemaPath      the path to the given schema
+   */
```

Comment:
I also have the same comment. Why not add getTables() to AbstractSchema interface and leave the logic of generating the info schema in RecordGenerator.